File: lib/core/models/summoning_type.dart
enum SummoningType {
  servant(summonCost: 4, controlCost: 2),
  djinn(summonCost: 8, controlCost: 4),
  master(summonCost: 12, controlCost: 8);

  final int summonCost;
  final int controlCost;
  const SummoningType({required this.summonCost, required this.controlCost});
}

File: pwa/src/core/models/summoning-type.ts
export enum SummoningType {
  Servant = 0,
  Djinn = 1,
  Master = 2,
}

export const SUMMONING_TYPE_COSTS: Record<SummoningType, { summon: number; control: number }> = {
  [SummoningType.Servant]: { summon: 4, control: 2 },
  [SummoningType.Djinn]: { summon: 8, control: 4 },
  [SummoningType.Master]: { summon: 12, control: 8 },
};
